### PR TITLE
Update to CLDR 40

### LINKED ISSUE #1187: Pull CLDR 40 into ICU4X
CLDR 40 has a big dump of new data as well as some improvements to the JSON generator.  We should try to get it in to 0.4.

### PR DESCRIPTION
**Pull request by sffc**
Fixes #1187

### CODE REVIEW

**Review by zbraniecki — DISMISSED** [review]
overall, lgtm! I hope you can use final 40 in the patch you land, and I left some request for more detail in why you're adding more locales to test data.

Other thoughts based on the PR:
* Super happy to see CldrLangId gone!
* Not sure how to reason about the `timeSkeletons`/`dateSkeletons` - Filed #1243 to discuss it.
* Seems like the number of keys `X-Y-Z` where `Y` is often `-count-` is increasing and we now have `masculine` and `feminine` for `Z` which is going to be *SUPER* tricky for CLDR to scale to locales with wide range of gender forms (Polish has different types and number of genders depneding on the plural form). Nonetheless, #1208 is filed to work on a more general logic.
* I'm a bit concerned about how quickly we're adding new locales to testdata - seems like with each feature we want to cover a new locale outlier to we just slap it into testdata. At this pace by 1.0 we'll have half of the locales in testdata :) Maybe it's ok to not have a locale in testdata for an edge case like `B` and just test it with synthetic pattern? Esp that if `zh-Hant` in CLDR 44 will remove `B` we are unlikely to understand and remember to remove `zh-Hant` from testdata ever.

File: provider/testdata/Cargo.toml [hunk 1]
```diff
@@ -98,6 +98,12 @@ locales = [
     # - Interesting case-mappings
     "tr",
 
+    # Chinese:
+    # - Multiple scripts
+    # - Complex date patterns
+    "zh",
+    "zh-Hant",
```
[the remark this answers is not in the document]
Comment (sffc) [reply provider/testdata/Cargo.toml:105]: acknowledged.  I added it in order to test the new `B` behavior, which is default in zh-Hant.  But I'll add a one-off test case for that instead of pulling in the whole locale.

File: provider/testdata/Cargo.toml [hunk 2]
```diff
@@ -123,7 +129,7 @@ cldr_json_glob = [
 ]
 
 # Git tag or sha1 for the CLDR data used to generate the testdata.
-gitref = "39.0.0"
+gitref = "40.0.0-BETA4d"
```
[the remark this answers is not in the document]
Comment (sffc) [reply provider/testdata/Cargo.toml]: Done.

File: provider/testdata/Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (sffc) [reply provider/testdata/Cargo.toml]: done.

**Review by zbraniecki — APPROVED** [review]
lgtm!
